### DISCUSSION TITLE
How to aggregate GTSdate for records table (Belchertown)

### DISCUSSION THREAD

**Discussion by silke-99** [roe-dl/weewx-GTS#31] [Q&A] [answered]
Hi, this might be a stupid question, but is there a way to do "alltime earliest GTSdate" aggregation in a simple way?

I played with this a little bit but can't wrap my head around how to do this.

Thanks for your help!

**Comment by roe-dl** [comment, marked as answer]
There are no stupid questions. This one is difficult. I looked at the aggregation types available in WeeWX, but I did not find anything appropriate. So I guess you need a little bit of Cheetah code. Something like this (not tested):

```
## find the earliest GTSdate
#set $mindaymonth = "12-31"
#set $mingtsdate = None
#for $yy in $alltime.years
#set $dt = $yy.GTSdate.last
#if $dt is not None and $dt.raw is not None
#set $daymonth = $dt.format("%m-%d")
#if $daymonth<$mindaymonth
#set $mindaymonth = $daymonth
#set $mingtsdate = $dt
#end if
#end if
#end for
$mingtsdate
```

This script involves a loop over all available years. For each year it checks whether `GTSdata` is available. If so, it formats a string containing of month and day of month. Please note, that it is important to have the month first. If this month and day value is less than the value found before, it is saved. If the loop is done, `$mingtsdate` contains the full date of the earliest `GTSdate` value.

**Reply by silke-99** [reply]
Very cool, this works! Thank you very much! :-)

**Reply by roe-dl** [reply]
Would you consider setting a star for the GTS extension? I assume those stars make it easier for people becoming aware of this extension.
